src/java.base/share/classes/java/util/UUID.java

```diff
@@ -469,34 +472,40 @@ public String toString() {
         long lsb = leastSigBits;
         long msb = mostSigBits;
         byte[] buf = new byte[36];
-        ByteArray.setLong(
+        UNSAFE.putLongUnaligned(
```

Comment:
I understood that since UUID was not used during startup, that using VarHandles is the preferred abstraction.

Reply:
Using VarHandle will be 10% slower than using Unsafe directly, I haven't found the reason yet

Reply:
The comment is already there, I think the value on the right is easier to understand in hexadecimal

Reply:
I believe it’s because `VarHandle`s do bounds checking:
https://github.com/openjdk/jdk/blob/581f90e242b8a943215a223189d171b7ede37785/src/java.base/share/classes/java/lang/invoke/X-VarHandleByteArrayView.java.template#L113-L116 https://github.com/openjdk/jdk/blob/581f90e242b8a943215a223189d171b7ede37785/src/java.base/share/classes/java/lang/invoke/X-VarHandleByteArrayView.java.template#L144-L161